app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,40 @@
 class PostsController < ApplicationController
-  def index; end
+  before_action :set_user
 
-  def show; end
+  def index
+    @posts = @user.posts
+  end
+
+  def show
+    @post = @user.posts.find(params[:id])
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.new(post_parameters)
+    @post.author_id = @user.id
+
+    if @post.save
+      flash[:notice] = 'The post is successfully submitted'
+      redirect_to user_post_path(@user, @post)
+    else
+      flash[:alert] = 'Can not save the post'
+      render :new
+    end
+  end
+
+  private
+
+  def set_user
+    @user = User.find(params[:user_id])
+  end
+
+  def post_parameters
+    params.require(:post).permit(:title, :text)
+  end
 end
```

Comment:
- [ ] As highligted earlier, these post parameters are the same with what you have in your schema.rb file 👇 

<img width="311" alt="Untitled" src="https://github.com/AlaaAlsalem/blogaapp/assets/103386183/0a91ecd2-de1d-4c3f-bf7e-80806b76624d">

**Please kindly look through your implementation and resolve any similar cases** 🙏